### ISSUE TITLE
How to define customize legend order with scanpy (sc.pl.umap)?

### ISSUE THREAD

**Issue by xueweireally** [scverse/scanpy#2290]
Hi, after check the parameters in sc.pl.umap, could you add a parameter 'legend_order' to customize legend order? Thanks.

**Comment by LisaSikkema** [comment]
You can actually already do this by changing the order of your categorical, e.g.:
```python
adata = sc.datasets.pbmc3k_processed()
# plot default:
sc.pl.umap(adata, color="louvain")
# reorder categories alphabetically
adata.obs.louvain = adata.obs.louvain.cat.reorder_categories(
    sorted(adata.obs.louvain.cat.categories)
)
# plot with new category order:
sc.pl.umap(adata, color="louvain")
```
Which gives:
<img width="390" alt="Screenshot 2022-09-24 at 19 07 31" src="https://user-images.githubusercontent.com/32548783/192110283-af0d14c5-0d79-4ecd-96ff-c079f5743887.png">

**Comment by LisaSikkema** [comment]
We could add it to the function docstring, not sure where would be best. `color` or `legend_loc` @Zethson ? Or better to add an argument that does the reordering for you ?

**Comment by xueweireally** [comment]
> You can actually already do this by changing the order of your categorical, e.g.:
> 
> ```python
> adata = sc.datasets.pbmc3k_processed()
> # plot default:
> sc.pl.umap(adata, color="louvain")
> # reorder categories alphabetically
> adata.obs.louvain = adata.obs.louvain.cat.reorder_categories(
>     sorted(adata.obs.louvain.cat.categories)
> )
> # plot with new category order:
> sc.pl.umap(adata, color="louvain")
> ```
> 
> Which gives: <img alt="Screenshot 2022-09-24 at 19 07 31" width="390" src="https://user-images.githubusercontent.com/32548783/192110283-af0d14c5-0d79-4ecd-96ff-c079f5743887.png">

Thanks for your replay.

Here, I changed the order of categorical as below:

# 0. loading data
adata = sc.datasets.pbmc3k_processed()
# 1. plot default:
sc.pl.umap(adata, color="louvain")

# 2. show the default order of categories:
adata.obs['louvain'].cat.categories
# **Index(['CD4 T cells', 'CD14+ Monocytes', 'B cells', 'CD8 T cells', 'NK cells', 'FCGR3A+ Monocytes', 'Dendritic cells', 'Megakaryocytes'], dtype='object')**

# 3. reorder categories as customize
adata.obs['batch_3rd'].cat.reorder_categories(['CD8 T cells', 'CD4 T cells', 'B cells', 'NK cells', 'CD14+ Monocytes', 'FCGR3A+ Monocytes', 'Dendritic cells', 'Megakaryocytes'], inplace=True, ordered=True)
adata.obs['batch_3rd'].cat.categories
# **Index(['CD8 T cells', 'CD4 T cells', 'B cells', 'NK cells', 'CD14+ Monocytes', 'FCGR3A+ Monocytes', 'Dendritic cells', 'Megakaryocytes'], dtype='object')**

# 4. plot with new category order:
sc.pl.umap(adata, color="louvain")

**Comment by xueweireally** [comment]
> We could add it to the function docstring, not sure where would be best. `color` or `legend_loc` @Zethson ? Or better to add an argument that does the reordering for you ?

I think `legend_loc` would be better. @Zethson

**Comment by Zethson** [comment]
I don't think that `legend_loc` would be the correct parameter. You are not even interacting with it during the example that @LisaSikkema posted (thanks for that btw!).

I see two options:
1. Adding a general statement to the `color` argument stating that reordering the categorical column to color by can customize the legend order
2. Adding your use-case to the example section in the pl.umap documentation.

Leaning more towards the second option.

What do you think? Would be great if you could attempt to submit a PR then. Happy to provide some guidance if required.
